### PR TITLE
1040 Indicate missing date for DiagnosticReport

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#1040

### Dependencies

none

### Description

- Indicate missing date for DiagnosticReport - [context](https://metriport.slack.com/archives/C0616FCPAKZ/p1724276023306869?thread_ts=1724257837.838079&cid=C0616FCPAKZ)
- Move some shared code to a shared file
- Include the resource ID (hidden) on DiagnosticReport and MedicationStatement

Note: I intentionally kept changes to a minimum. There's a lot to fix/improve, but this needs more time and prioritization.

### Testing

- Local
  - [x] Generate regular MR
  - [x] Generate ADHD MR
- Staging
  - [ ] Generate regular MR
  - [ ] Generate ADHD MR
- Sandbox
  - none
- Production
  - none

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 1]
```diff
@@ -684,22 +683,20 @@ function createFilteredReportSection(
...
+      const encounterDate = key;
       const conditionVisit = filterConditions.find(condition => {
         const conditionId = condition.id ?? "";
         const conditionVisitDate = condition.onsetDateTime
           ? condition.onsetDateTime
           : conditionDateDict[conditionId]?.start;
 
-        const conditionVisitDateFormatted = dayjs(conditionVisitDate).format(ISO_DATE);
-
-        return conditionVisitDateFormatted === encounterDateFormatted;
```
Comment (leite08) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:702]: `key` already is a string w/ date in ISO format, no need to convert again

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 2]
```diff
@@ -738,67 +735,6 @@ function createFilteredReportSection(
   `;
 }
 
-function buildEncounterSections(
-  encounterSections: EncounterSection,
-  diagnosticReports: DiagnosticReport[]
-): EncounterSection {
```
Comment (leite08) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:744]: moved to the shared file, both regular and ADHD MRs had the same code

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 3]
```diff
@@ -1096,7 +1034,7 @@ function createWhatWasDocumentedFromDiagnosticReports(
         field => field.trim().length > 0
       );
       return `
-      <div>
+      <div data-id="${doc.id}">
```
Comment (leite08) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:1037]: Adding the DiagnosticReport's ID to the MR to help w/ debugging

File: packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts [hunk 4]
```diff
@@ -1305,19 +1244,20 @@ function createConditionSection(conditions: Condition[], encounter: Encounter[])
...
+      let onsetStartTime = condition.onsetPeriod?.start;
+      let onsetEndTime = condition.onsetPeriod?.end;
 
       if (!onsetStartTime && condition.id) {
-        onsetStartTime = conditionDateDict[condition.id]?.start ?? "";
+        onsetStartTime = conditionDateDict[condition.id]?.start;
       }
       if (!onsetEndTime && condition.id) {
-        onsetEndTime = conditionDateDict[condition.id]?.end ?? "";
+        onsetEndTime = conditionDateDict[condition.id]?.end;
```
Comment (leite08) [inline packages/core/src/external/aws/lambda-logic/bundle-to-html-adhd.ts:1256]: If those were not set, we'd be sending empty string to dayjs' constructor.

**Review by thomasyopes — APPROVED** [review]

**Pull request merged**